org.eclipse.draw2d/src/org/eclipse/draw2d/FigureCanvas.java

```diff
@@ -300,9 +300,17 @@ private void unhookViewport() {
 
 	private void layoutViewport() {
 		ScrollPaneSolver.Result result;
+		int vBarWidth = 0;
+		int hBarHeight = 0;
+
+		if ((getScrollbarsMode() & SWT.SCROLLBAR_OVERLAY) != 0) {
```

Comment:
We got this check wrong... `computeTrim(...)` should be called when scrollbar overlay is disabled (i.e. == 0), not the other way around.

Reply:
Oups. You are right. What's the good way to fix this? Do I need to create another pull request?

Reply:
I've already created https://github.com/eclipse/gef-classic/pull/429